### PR TITLE
BUGFIX: SD-833 set min value for radius circle with Null value

### PR DESCRIPTION
**Pull request by dlcd2203**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2855
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2597

# Related Jira tasks

SD-833: https://jira.intgdc.com/browse/SD-833

### CODE REVIEW

**Comment by dlcd2203** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6475/ : SUCCESS in 11m 43s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2263/ : SUCCESS in 7s

**Comment by dlcd2203** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by dlcd2203** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3363/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2266/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2764/ : UNSTABLE in 18m 23s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6476/ : SUCCESS in 19m 35s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2265/ : SUCCESS in 7s

**Comment by dlcd2203** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6480/ : SUCCESS in 6m 14s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2273/ : SUCCESS in 6s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3366/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2274/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2768/ : UNSTABLE in 15m 13s

**Comment by dlcd2203** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1436.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2275/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2769/ : SUCCESS in 25m 02s

File: src/components/core/geoChart/geoChartDataLayers.ts [hunk 1]
```diff
@@ -46,17 +46,21 @@ function createPushpinSizeOptions(geoData: IGeoData): mapboxgl.Expression | numb
     // The mouseenter event uses queryRenderedFeatures to determine when the mouse is touching a feature
     // And queryRenderedFeatures is not supporting nested object in expression
     // https://github.com/mapbox/mapbox-gl-js/issues/7194
-    const sizeOptions: mapboxgl.Expression = ["step", ["get", "pushpinRadius"], DEFAULT_PUSHPIN_SIZE_VALUE];
+
+    const stepSizeOptions: mapboxgl.Expression = [
```
Comment (ptquang86) [inline src/components/core/geoChart/geoChartDataLayers.ts]: Please use the name `sizeOptions` instead :)

File: src/components/core/geoChart/geoChartDataLayers.ts [hunk 2]
```diff
@@ -46,17 +46,21 @@ function createPushpinSizeOptions(geoData: IGeoData): mapboxgl.Expression | numb
...
         stepValue = parseFloat(stepValue.toFixed(2));
         if (stepValue > sizeMax) {
             stepValue = sizeMax;
         }
-        sizeOptions.push(stepValue, DEFAULT_PUSHPIN_SIZE_SCALE[index]);
+        stepSizeOptions.push(stepValue, DEFAULT_PUSHPIN_SIZE_SCALE[index]);
     }
-
-    return sizeOptions;
+    return ["case", ["==", ["get", "pushpinRadius"], null], DEFAULT_PUSHPIN_SIZE_VALUE, stepSizeOptions];
```
Comment (ptquang86) [inline src/components/core/geoChart/geoChartDataLayers.ts]: should we care about `undefined`, `NaN`?

File: src/components/core/geoChart/tests/geoChartDataLayers.spec.ts [hunk 3]
```diff
@@ -121,21 +121,26 @@ describe("createPushpinDataLayer", () => {
             "rgb(20,178,226)",
         ]);
         expect(layer.paint["circle-radius"]).toEqual([
-            "step",
-            ["get", "pushpinRadius"],
+            "case",
+            ["==", ["get", "pushpinRadius"], null],
             4,
```
Comment (ptquang86) [inline src/components/core/geoChart/tests/geoChartDataLayers.spec.ts]: Why is `4` here?

Comment (dlcd2203) [reply src/components/core/geoChart/tests/geoChartDataLayers.spec.ts]: set min size (4) for Null value

File: src/components/core/geoChart/geoChartDataLayers.ts (hunk 2 above)
Comment (dlcd2203) [reply src/components/core/geoChart/geoChartDataLayers.ts]: Currently, data in feature of Mapbox is `NaN`. And I tested, filter Mapbox just works with comparing  `null`  value.

File: src/components/core/geoChart/geoChartDataLayers.ts (hunk 1 above)
Comment (dlcd2203) [reply src/components/core/geoChart/geoChartDataLayers.ts]: fixed

**Comment by ptquang86** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6486/ : SUCCESS in 16m 08s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2280/ : SUCCESS in 6s

File: src/components/core/geoChart/geoChartDataLayers.ts [hunk 4]
```diff
@@ -46,6 +46,7 @@ function createPushpinSizeOptions(geoData: IGeoData): mapboxgl.Expression | numb
     // The mouseenter event uses queryRenderedFeatures to determine when the mouse is touching a feature
     // And queryRenderedFeatures is not supporting nested object in expression
     // https://github.com/mapbox/mapbox-gl-js/issues/7194
+
```
Comment (naibinh) [inline src/components/core/geoChart/geoChartDataLayers.ts]: We should remove this line, this explanation aims to `["step", ["get", "pushpinRadius"], DEFAULT_PUSHPIN_SIZE_VALUE]`

Comment (dlcd2203) [reply src/components/core/geoChart/geoChartDataLayers.ts]: fixed

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6506/ : SUCCESS in 12m 59s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2305/ : SUCCESS in 8s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6530/ : FAILURE in 10m 50s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2332/ : SUCCESS in 5s

**Comment by dlcd2203** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6532/ : SUCCESS in 5m 39s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2334/ : SUCCESS in 4s

**Comment by naibinh** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3389/ : SUCCESS in 17m 49s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6534/ : SUCCESS in 21m 40s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1574/ : SUCCESS in 2m 30s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2263/ : SUCCESS in 22m 43s

**Pull request merged**